docs/reST/ref/camera.rst

```diff
@@ -8,7 +8,11 @@
 
 | :sl:`pygame module for camera use`
 
-Pygame currently supports only Linux and v4l2 cameras.
+Pygame currently supports Linux (v4l2) and Windows (MSMF) cameras.
```

Comment:
I guess it can also use opencv or VideoCapture if they are installed.

Reply:
I think the opencv one uses a different python port of opencv, with different imports / names than the opencv commonly used today (`cv2`). I've actually written a new opencv pygame camera backend that I want to replace the old one with, I just want to get this merged before I deal with the buildconfig changes for that.

VideoCapture is py2 only, windows only, and was last updated in 2012, so I'm hoping to support that more as a legacy thing.

Reply:
Cool! I forgot it changed API.
I used cv2 in here: https://github.com/pygame/parrotjoy/blob/master/parrotjoy/camera_cv.py

Note, how it uses a thread queue. I was wondering about that sort of interface where you get an event on the pygame event queue with the surface. The python queue is latency inducing, which is the reason for using the pygame one. The nice thing about using the queue is it sort of makes it easier to know when a new frame comes in - and it doesn't block.

Another idea was it might be that rather than allocating a frame every time it could reuse surfaces. It would be good if we could get rid of the extra rotate/flip stuff. Note in my one it's using the opencv code for those rotate things rather than numpy... but maybe there's a way to avoid them all together.

There's newer VideoCapture modules available for py3 at the unofficial windows binaries page: https://www.lfd.uci.edu/~gohlke/pythonlibs/#videocapture (along with lots of other binaries often not available elsewhere on windows). gohlke has occasionally contributed to pygame in the past, and maybe has the code available with some py3 changes if you're interested.

Reply:
>  It would be good if we could get rid of the extra rotate/flip stuff. Note in my one it's using the opencv code for those rotate things rather than numpy... but maybe there's a way to avoid them all together.

In my research I didn't find any other way to tell opencv to do mirroring.

> There's newer VideoCapture modules available for py3 at the unofficial windows binaries page: https://www.lfd.uci.edu/~gohlke/pythonlibs/#videocapture

Very interesting. I looked at the VideoCapture source and it doesn't look like it should support python3, I wonder if gohlke has his own set of patches. A third party has uploaded files to Pypi for it: https://pypi.org/project/VideoCapture/0.9.5/.
However, I get `Cannot set capture resolution.` when I try to run pygame.camera with it.

Reply:
I guess there must be some other patches for it. I'll 'Resolve conversation' on here though, because that's sort of off topic for this PR.